zcash_client_sqlite/src/wallet.rs

```diff
@@ -3030,6 +3030,7 @@ pub(crate) fn set_transaction_status(
             conn.execute(
                 "UPDATE transactions
                  SET mined_height = :height,
+                     min_observed_height = MIN(IFNULL(mined_height, :height), :height),
```

Comment:
Bug: this ignores the current value of `min_observed_height` and can cause it to increase. We need the minimum of all three values.